modules/juce_audio_devices/native/juce_android_OpenSL.cpp

```diff
@@ -620,21 +643,26 @@ class OpenSLAudioIODevice  : public AudioIODevice,
                                                                    static_cast<SLuint32> (bufferList.numBuffers) };
             SLDataSink audioSink = { &bufferQueue, &pcmFormat };
 
-            const SLInterfaceID interfaceIDs[] = { *engine.SL_IID_ANDROIDSIMPLEBUFFERQUEUE };
+            const SLInterfaceID interfaceIDs[] = { *engine.SL_IID_ANDROIDSIMPLEBUFFERQUEUE,
+                *engine.SL_IID_ANDROIDCONFIGURATION };

```

Comment:
The `SL_IID_ANDROIDCONFIGURATION` interface provides the `SetConfiguration` method, which is used in `Recorder::setAudioPreprocessingEnabled()`. If we create the audio recorder without that interface, `Recorder::configObject` (line 657) will be `NULL` and `Recorder::setAudioPreprocessingEnabled()` will fail. It looks like a bug in JUCE... Superpowered does instantiate the audio recorder with `SL_IID_ANDROIDCONFIGURATION` enabled.